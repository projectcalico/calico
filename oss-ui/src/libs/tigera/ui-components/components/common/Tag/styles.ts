import { transparentize } from '@chakra-ui/theme-tools';

export default {
    baseStyle: {},
    variants: {
        subtle: {
            container: {
                bg: 'tigeraGrey.200',
                borderRadius: 'lg',
                fontStyle: 'normal',
                fontWeight: 'normal',
                fontSize: 'xxs',
                lineHeight: '1',
                height: '4',
                minHeight: '4',
                px: '4',
                color: 'tigeraBlack',
                _dark: {
                    color: 'tigeraGrey.100',
                },
            },
        },
    },
};

const inactiveStyles = {
    color: transparentize(`tigeraBlack`, 0.5),
};
export { inactiveStyles };
