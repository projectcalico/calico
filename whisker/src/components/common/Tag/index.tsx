import {
    Tag as ChakraTag,
    TagCloseButton,
    TagProps as ChakraTagProps,
} from '@chakra-ui/react';

type Tag = {
    value: string;
    label: string;
};
export type TagProps = ChakraTagProps & {
    tag: Tag;
    onRemove: (option: Tag) => void;
};

const Tag: React.FC<TagProps> = ({ tag, onRemove, ...rest }) => (
    <ChakraTag variant='solid' cursor='default' {...rest}>
        {tag.label}
        <TagCloseButton
            onClick={(event) => {
                event.stopPropagation();
                onRemove(tag);
            }}
        />
    </ChakraTag>
);

export default Tag;
