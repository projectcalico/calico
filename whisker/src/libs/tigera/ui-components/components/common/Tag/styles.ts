import { transparentize } from '@chakra-ui/theme-tools';

export default {
    baseStyle: {},
    variants: {
        subtle: {
            container: {
                bg: 'tigeraGrey.200',
                borderRadius: 'lg',
                fontStyle: 'normal',
                fontWeight: 'normal',
                fontSize: 'xxs',
                lineHeight: '1',
                height: '4',
                minHeight: '4',
                px: '4',
                color: 'tigeraBlack',
                _dark: {
                    color: 'tigeraGrey.100',
                },
            },
        },
        solid: {
            container: {
                bg: 'tigeraGrey.600',
                borderRadius: 'sm',
                lineHeight: '1',
                minHeight: '1',
                fontWeight: 'normal',
                fontSize: 'sm',
                pr: 0,
                gap: 1,
            },
            closeButton: {
                m: 0,
                fontSize: 'sm',
                borderRadius: 'none',
                height: '22px',
                _hover: {
                    bg: 'tigeraGrey.400',
                    color: 'tigeraBlack',
                },
            },
        },
    },
};

const inactiveStyles = {
    color: transparentize(`tigeraBlack`, 0.5),
};
export { inactiveStyles };
