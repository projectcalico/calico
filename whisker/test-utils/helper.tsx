import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
