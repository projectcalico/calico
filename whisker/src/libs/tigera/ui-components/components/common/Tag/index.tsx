import * as React from 'react';
import { Tag as ChakraTag, TagProps } from '@chakra-ui/react';
import { inactiveStyles } from './styles';

const Tag: React.FC<
    React.PropsWithChildren<TagProps & { isActive?: boolean }>
> = ({ isActive = true, sx, ...rest }) => {
    return (
        <ChakraTag
            sx={{ ...(!isActive ? inactiveStyles : {}), ...sx }}
            {...rest}
        />
    );
};

export default Tag;
