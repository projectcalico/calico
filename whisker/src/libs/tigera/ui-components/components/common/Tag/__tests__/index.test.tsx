import { render } from '@/test-utils/helper';
import Tag from '../index';

describe('<Tag/>', () => {
    it('it renders', () => {
        const { asFragment } = render(<Tag>Hola active tag!</Tag>);

        expect(asFragment()).toMatchSnapshot();
    });

    it('it renders inactive', () => {
        const { asFragment } = render(
            <Tag isActive={false}>Hola inactive tag!</Tag>,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
