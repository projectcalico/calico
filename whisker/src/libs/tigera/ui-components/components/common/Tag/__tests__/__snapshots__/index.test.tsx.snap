// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag/> it renders 1`] = `
<DocumentFragment>
  <span
    class="css-16748on"
  >
    Hola active tag!
  </span>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Tag/> it renders inactive 1`] = `
<DocumentFragment>
  <span
    class="css-7dr3bu"
  >
    Hola inactive tag!
  </span>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
